### ISSUE TITLE
Support k8s 1.5

### ISSUE THREAD

**Issue by ivan4th** [Mirantis/virtlet#154]
[Currently](https://github.com/kubernetes/kubernetes/commit/ad1c60da71ff59b17df8a567390153771370cbbf) this means also support for k8s 1.5

**Comment by pigmej** [comment]
I'd say that we should support 1.5 in first place then 1.6 ;)

**Comment by ivan4th** [comment]
`pkg/kubelet/api/v1alpha1/runtime/api.proto` without comments, diff between v1.5.0-alpha.1 and [current version]((https://github.com/kubernetes/kubernetes/blob/ad1c60da71ff59b17df8a567390153771370cbbf/pkg/kubelet/api/v1alpha1/runtime/api.proto):

```
--- /tmp/api.proto.old	2016-12-13 12:33:08.000000000 +0300
+++ /tmp/api.proto.new	2016-12-13 12:33:28.000000000 +0300
@@ -18,7 +18,14 @@
     rpc ListContainers(ListContainersRequest) returns (ListContainersResponse) {}
     rpc ContainerStatus(ContainerStatusRequest) returns (ContainerStatusResponse) {}
 
-    rpc Exec(stream ExecRequest) returns (stream ExecResponse) {}
+    rpc ExecSync(ExecSyncRequest) returns (ExecSyncResponse) {}
+    rpc Exec(ExecRequest) returns (ExecResponse) {}
+    rpc Attach(AttachRequest) returns (AttachResponse) {}
+    rpc PortForward(PortForwardRequest) returns (PortForwardResponse) {}
+
+    rpc UpdateRuntimeConfig(UpdateRuntimeConfigRequest) returns (UpdateRuntimeConfigResponse) {}
+
+    rpc Status(StatusRequest) returns (StatusResponse) {}
 }
 
 service ImageService {
@@ -70,9 +77,17 @@
     optional bool host_ipc = 3;
 }
 
+message LinuxSandboxSecurityContext {
+    optional NamespaceOption namespace_options = 1;
+    optional SELinuxOption selinux_options = 2;
+    optional int64 run_as_user = 3;
+    optional bool readonly_rootfs = 4;
+    repeated int64 supplemental_groups = 5;
+}
+
 message LinuxPodSandboxConfig {
     optional string cgroup_parent = 1;
-    optional NamespaceOption namespace_options = 2;
+    optional LinuxSandboxSecurityContext security_context = 2;
 }
 
 message PodSandboxMetadata {
@@ -130,15 +145,15 @@
     optional Namespace namespaces = 1;
 }
 
-enum PodSandBoxState {
-    READY = 0;
-    NOTREADY = 1;
+enum PodSandboxState {
+    SANDBOX_READY = 0;
+    SANDBOX_NOTREADY = 1;
 }
 
 message PodSandboxStatus {
     optional string id = 1;
     optional PodSandboxMetadata metadata = 2;
-    optional PodSandBoxState state = 3;
+    optional PodSandboxState state = 3;
     optional int64 created_at = 4;
     optional PodSandboxNetworkStatus network = 5;
     optional LinuxPodSandboxStatus linux = 6;
@@ -152,7 +167,7 @@
 
 message PodSandboxFilter {
     optional string id = 1;
-    optional PodSandBoxState state = 2;
+    optional PodSandboxState state = 2;
     map<string, string> label_selector = 3;
 }
 
@@ -164,7 +179,7 @@
 message PodSandbox {
     optional string id = 1;
     optional PodSandboxMetadata metadata = 2;
-    optional PodSandBoxState state = 3;
+    optional PodSandboxState state = 3;
     optional int64 created_at = 4;
     map<string, string> labels = 5;
     map<string, string> annotations = 6;
@@ -203,17 +218,20 @@
     repeated string drop_capabilities = 2;
 }
 
-message LinuxContainerConfig {
-    optional LinuxContainerResources resources = 1;
-    optional Capability capabilities = 2;
-    optional SELinuxOption selinux_options = 3;
-    optional LinuxUser user = 4;
+message LinuxContainerSecurityContext {
+    optional Capability capabilities = 1;
+    optional bool privileged = 2;
+    optional NamespaceOption namespace_options = 3;
+    optional SELinuxOption selinux_options = 4;
+    optional int64 run_as_user = 5;
+    optional string run_as_username = 6;
+    optional bool readonly_rootfs = 7;
+    repeated int64 supplemental_groups = 8;
 }
 
-message LinuxUser {
-    optional int64 uid = 1;
-    optional int64 gid = 2;
-    repeated int64 additional_gids = 3;
+message LinuxContainerConfig {
+    optional LinuxContainerResources resources = 1;
+    optional LinuxContainerSecurityContext security_context = 2;
 }
 
 message ContainerMetadata {
@@ -221,6 +239,12 @@
     optional uint32 attempt = 2;
 }
 
+message Device {
+    optional string container_path = 1;
+    optional string host_path = 2;
+    optional string permissions = 3;
+}
+
 message ContainerConfig {
     optional ContainerMetadata metadata = 1 ;
     optional ImageSpec image = 2;
@@ -229,17 +253,16 @@
     optional string working_dir = 5;
     repeated KeyValue envs = 6;
     repeated Mount mounts = 7;
-    map<string, string> labels = 8;
-    map<string, string> annotations = 9;
-    optional bool privileged = 10;
-    optional bool readonly_rootfs = 11;
-    optional string log_path = 12;
-
-    optional bool stdin = 13;
-    optional bool stdin_once = 14;
-    optional bool tty = 15;
+    repeated Device devices = 8;
+    map<string, string> labels = 9;
+    map<string, string> annotations = 10;
+    optional string log_path = 11;
+
+    optional bool stdin = 12;
+    optional bool stdin_once = 13;
+    optional bool tty = 14;
 
-    optional LinuxContainerConfig linux = 16;
+    optional LinuxContainerConfig linux = 15;
 }
 
 message CreateContainerRequest {
@@ -272,10 +295,10 @@
 message RemoveContainerResponse {}
 
 enum ContainerState {
-    CREATED = 0;
-    RUNNING = 1;
-    EXITED  = 2;
-    UNKNOWN = 3;
+    CONTAINER_CREATED = 0;
+    CONTAINER_RUNNING = 1;
+    CONTAINER_EXITED  = 2;
+    CONTAINER_UNKNOWN = 3;
 }
 
 message ContainerFilter {
@@ -330,16 +353,46 @@
     optional ContainerStatus status = 1;
 }
 
+message ExecSyncRequest {
+    optional string container_id = 1;
+    repeated string cmd = 2;
+    optional int64 timeout = 3;
+}
+
+message ExecSyncResponse {
+    optional bytes stdout = 1;
+    optional bytes stderr = 2;
+    optional int32 exit_code = 3;
+}
+
 message ExecRequest {
     optional string container_id = 1;
     repeated string cmd = 2;
     optional bool tty = 3;
-    optional bytes stdin = 4;
+    optional bool stdin = 4;
 }
 
 message ExecResponse {
-    optional bytes stdout = 1;
-    optional bytes stderr = 2;
+    optional string url = 1;
+}
+
+message AttachRequest {
+    optional string container_id = 1;
+    optional bool stdin = 2;
+    optional bool tty = 3;
+}
+
+message AttachResponse {
+    optional string url = 1;
+}
+
+message PortForwardRequest {
+    optional string pod_sandbox_id = 1;
+    repeated int32 port = 2;
+}
+
+message PortForwardResponse {
+    optional string url = 1;
 }
 
 message ImageFilter {
@@ -355,6 +408,8 @@
     repeated string repo_tags = 2;
     repeated string repo_digests = 3;
     optional uint64 size = 4;
+    optional int64 uid = 5;
+    optional string username = 6;
 }
 
 message ListImagesResponse {
@@ -391,3 +446,34 @@
 }
 
 message RemoveImageResponse {}
+
+message NetworkConfig {
+    optional string pod_cidr = 1;
+}
+
+message RuntimeConfig {
+    optional NetworkConfig network_config = 1;
+}
+
+message UpdateRuntimeConfigRequest {
+    optional RuntimeConfig runtime_config = 1;
+}
+
+message UpdateRuntimeConfigResponse {}
+
+message RuntimeCondition {
+    optional string type = 1;
+    optional bool status = 2;
+    optional string reason = 3;
+    optional string message = 4;
+}
+
+message RuntimeStatus {
+    repeated RuntimeCondition conditions = 1;
+}
+
+message StatusRequest {}
+
+message StatusResponse {
+    optional RuntimeStatus status = 1;
+}
```

**Comment by jellonek** [comment]
Closed by #170
